pkg/doc/jose/encrypter.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package jose
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/google/tink/go/keyset"
+	"github.com/google/tink/go/subtle/hybrid"
+	"github.com/square/go-jose/v3"
+
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/api"
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/ecdhes"
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/ecdhes/subtle"
+)
+
+// Encrypter interface to Encrypt/Decrypt JWE messages
+type Encrypter interface {
+	// Encrypt plaintext and aad sent to 1 or more recipients and return a valid JSONWebEncryption instance
+	Encrypt(plaintext, aad []byte) (*JSONWebEncryption, error)
+	// Decrypt a marshalledJWE, extract the corresponding recipient key to decrypt plaintext and return it
+	Decrypt(marshalledJWE []byte) ([]byte, error)
+}
+
+// JWEEncrypt is responsible of encrypting a plaintext and it's aad into a protected JWE and decrypting it
+type JWEEncrypt struct {
+	recipients []subtle.ECPublicKey
+	senderKH   *keyset.Handle
+}
+
+// NewJWEEncrypt creates a new JWEEncrypt instance to build/prase JWE with recipientsPubKeys
+func NewJWEEncrypt(recipientsPubKeys []subtle.ECPublicKey) (*JWEEncrypt, error) {
+	if len(recipientsPubKeys) == 0 {
+		return nil, fmt.Errorf("empty recipientsPubKeys list")
+	}
+
+	kt, err := ecdhes.ECDHES256KWAES256GCMKeyTemplateWithRecipients(recipientsPubKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	senderKH, err := keyset.NewHandle(kt)
+	if err != nil {
+		return nil, err
+	}
+
+	return &JWEEncrypt{
+		recipients: recipientsPubKeys,
+		senderKH:   senderKH,
+	}, nil
+}
+
+func (je *JWEEncrypt) getEncryptionPrimitive() (api.CompositeEncrypt, error) {
+	senderPubKH, err := je.senderKH.Public()
+	if err != nil {
+		return nil, err
+	}
+
+	return ecdhes.NewECDHESEncrypt(senderPubKH)
+}
+
+// Encrypt plaintext with aad and return a JSONWebEncryption instance to serialize JWE instance
```

Comment:
I'd suggest capitalizing AAD in the comment

Reply:
updated, thank you